src/map/packet_system.cpp

```diff
@@ -7609,6 +7609,19 @@ void SmallPacket0x102(map_session_data_t* const PSession, CCharEntity* const PCh
             {
                 ShowDebug("No match found. ");
             }
+
+            // Regardless what the set spell action is, force recast on all currently-set blu spells
+            for (uint8 i = 0; i < 20; i++)
+            {
+                if (PChar->m_SetBlueSpells[i] != 0)
+                {
+                    CBlueSpell* BlueSpell = (CBlueSpell*)spell::GetSpell(static_cast<SpellID>(PChar->m_SetBlueSpells[i] + 0x200));
```

Comment:
The check for BlueSpell being nullptr is only likely to work if GetSpell returns nullptr, if it returns an incompatible spell then it'll succeed and strange things may happen. This is a use case for `dynamic_cast`:

```cpp
auto  spellId = static_cast<SpellID>(PChar->m_SetBlueSpells[i] + 0x200);
auto* PSpell  = spell::GetSpell(spellId);
if (CBlueSpell* PBlueSpell = dynamic_cast<CBlueSpell*>(PSpell))
{
    PChar->PRecastContainer->Add(RECAST_MAGIC, static_cast<uint16>(PBlueSpell->getID()), 60);
}
````

Also, prefer static cast to c-style casting

Reply:
Changes made, also noticed it was accidentally in the `then` section only applying it when you _set_ a spell, not when you _remove_ a spell.

Tested and confirmed both cases lock out blu spells for 60s

Thanks!